### PR TITLE
feat(choreo): fork choice

### PR DESCRIPTION
**Pull request by lidatong**

### CODE REVIEW

File: src/waltz/fd_waltz.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef HEADER_fd_src_waltz_fd_waltz_h
+#define HEADER_fd_src_waltz_fd_waltz_h
```
[the remark this answers is not in the document]
Comment (lidatong) [reply src/waltz/fd_waltz.h:2]: oh i didnt realize you had claimed the name `waltz`. i just picked another name. happy to pick another one

[the remark this answers is not in the document]
Comment (lidatong) [reply src/waltz/fd_waltz.h:2]: encourage modularity i.e. consensus should be able to run and tested independently, avoid entanglement with replay

**Comment by mmcgee-jump** [comment]
Does this really need to be a new lib?

**Comment by mmcgee-jump** [comment]
If we followed this lib convention in other areas of the code we would have one, for, eg, shredding, poh, etc. I think it's fine in ballet or flamenco.

**Comment by lidatong** [comment]
> If we followed this lib convention in other areas of the code we would have one, for, eg, shredding, poh, etc. I think it's fine in ballet or flamenco.

agree we shouldnt proliferate libs but imo consensus should live on its own so that it can be built, run, etc. independently without any dependency on flamenco / the rest of runtime

not just for modularity but for ease of testing especially e.g. making it easier for external auditors to use

**Pull request closed without merging**